node/src/main/scala/coop/rchain/node/service/jsonrpc.scala

```diff
@@ -0,0 +1,128 @@
+package coop.rchain.node.service
+
+import cats.effect._
+import io.circe._
+import io.circe.generic.auto._
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras._
+import io.circe.literal._
+import io.circe.syntax._
+import org.http4s._
+import org.http4s.circe._
+import org.http4s.dsl.io._
+
+import shapeless._
+
+object JsonRpc {
+  implicit val config: Configuration = Configuration.default.withSnakeCaseMemberNames
+
+  type IdType = Long :+: String :+: CNil
+
+  implicit val encodeIdType: Encoder[IdType] = new Encoder[IdType] {
+    final def apply(x: IdType): Json =
+      x match {
+        case Inl(i)      => Json.fromLong(i)
+        case Inr(Inl(s)) => Json.fromString(s)
+        case _           => ???
+      }
+  }
```

Comment:
Still seems like there should be a better way to do this.  Perhaps @rabbitonweb knows the ways of the shapeless.  There's also a `circe-shapes` package that might come in handy for this purpose.